src/main/webapp/scripts/controlGameInstance.js

```diff
@@ -155,17 +155,18 @@ function initUIButtons(gameInstanceId) {
   const endGameInstanceButton = document.getElementById("endGameInstanceButton");
   
   startGameInstanceButtonElement.addEventListener('click', () => {
-      fetch('/startGameInstance?action=start&gameInstance='+activeGameInstanceId);
+      fetch('/startGameInstance?gameInstance='+activeGameInstanceId, { method: 'POST' });
   });
   nextQuestionButton.addEventListener('click', () => {
-      fetch('/startGameInstance?action=next&gameInstance='+activeGameInstanceId);
+      fetch('/nextQuestion?gameInstance='+activeGameInstanceId, { method: 'POST' });
   });
   previousQuestionButton.addEventListener('click', () => {
-      fetch('/startGameInstance?action=previous&gameInstance='+activeGameInstanceId);
+      fetch('/previousQuestion?gameInstance='+activeGameInstanceId, { method: 'POST' });
   });
   endGameInstanceButton.addEventListener('click', () => {
-      fetch('/startGameInstance?action=end&gameInstance='+activeGameInstanceId);
-      window.location.href = "/teacher/controlGameInstance.html";
+      fetch('/endGameInstance?gameInstance='+activeGameInstanceId, { method: 'POST' }).then(()=>{
+          window.location.href = "/teacher/controlGameInstance.html";
```

Comment:
So when the teacher ends the game right now it just resets the page as if the game had never happened right?

Reply:
Yes exactly, it is as the initial state before the start.